expression/builtin_like_vec.go

```diff
@@ -27,17 +30,109 @@ func (b *builtinLikeSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
 }
 
 func (b *builtinRegexpBinarySig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRegexpBinarySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	bufExpr, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufExpr)
+	if err := b.args[0].VecEvalString(b.ctx, input, bufExpr); err != nil {
+		return err
+	}
+
+	bufPat, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufPat)
+	if err := b.args[1].VecEvalString(b.ctx, input, bufPat); err != nil {
+		return err
+	}
+
+	if isRegexpMemoizable(b.args[1]) && b.rm == nil {
+		b.rm = newRegexpMemoizer(bufPat, n, b)
+	}
+	getRegexp := func(pat string) (*regexp.Regexp, error) {
+		if b.rm != nil && b.rm.re != nil {
+			return b.rm.re, nil
+		}
+		return b.compile(pat)
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(bufExpr, bufPat)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		re, err := getRegexp(bufPat.GetString(i))
+		if err != nil {
+			return err
+		}
+		i64s[i] = boolToInt64(re.MatchString(bufExpr.GetString(i)))
+	}
+	return nil
 }
 
 func (b *builtinRegexpSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRegexpSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
```

Comment:
What is the difference between `builtinRegexpSig ` and `builtinRegexpBinarySig `? I see the two `vecEvalInt()` are almos the same.

Reply:
Ack, created a `builtinRegexpSharedSig` to reuse code